Code/GraphMol/SmilesParse/SmilesParse.cpp

```diff
@@ -422,13 +422,13 @@ RWMol *SmilesToMol(const std::string &smiles,
       throw;
     }
     // figure out stereochemistry:
-    if (params.useLegacyStereo) {
-      bool cleanIt = true, force = true, flagPossible = true;
-      MolOps::assignStereochemistry(*res, cleanIt, force, flagPossible);
-    } else {
-      bool cleanIt = true, flagPossible = false;
-      Chirality::findPotentialStereo(*res, cleanIt, flagPossible);
+    bool oVal = Chirality::getUseLegacyStereoPerception();
+    if (!params.useLegacyStereo) {
+      Chirality::setUseLegacyStereoPerception(false);
     }
+    bool cleanIt = true, force = true, flagPossible = true;
+    MolOps::assignStereochemistry(*res, cleanIt, force, flagPossible);
```

Comment:
I don't see any obvious exceptions in `assignStereochemistry()`, but if it did throw, the global stereo perception state would get mucked up.

I guess this could also cause concurrency issues, but I think that those are more theoretical, because it would be weird to have different concurrent tasks that would want different stereo perception protocols.